src/errors/custom.errors.ts

```diff
@@ -0,0 +1,36 @@
+import { CUSTOM_ERRORS } from '@/enums'
+
+export class UnknownCurrencyError extends Error {
+  constructor() {
+    super()
+    this.name = CUSTOM_ERRORS.unknownCurrency
```

Comment:
Don't quite get the purpose of `CUSTOM_ERRORS` enum.

do you use `name` somewhere?